controller/test/blower_fsm/blower_fsm_test.cpp

```diff
@@ -73,12 +75,30 @@ TEST_F(BlowerFsmTest, PressureControl) {
   p.pip_cm_h2o = 20;
 
   testSequence({
-      {p, 0, cmH2O(20), ValveState::CLOSED},
-      {p, 1000, cmH2O(20), ValveState::CLOSED},
-      {p, 1999, cmH2O(20), ValveState::CLOSED},
-      {p, 2001, cmH2O(10), ValveState::OPEN},
-      {p, 2999, cmH2O(10), ValveState::OPEN},
-      {p, 3001, cmH2O(20), ValveState::CLOSED},
+      {p, /*blower_enabled=*/true, 0, cmH2O(20), ValveState::CLOSED},
+      {p, /*blower_enabled=*/true, 1000, cmH2O(20), ValveState::CLOSED},
+      {p, /*blower_enabled=*/true, 1999, cmH2O(20), ValveState::CLOSED},
+      {p, /*blower_enabled=*/true, 2001, cmH2O(10), ValveState::OPEN},
+      {p, /*blower_enabled=*/true, 2999, cmH2O(10), ValveState::OPEN},
+      {p, /*blower_enabled=*/true, 3001, cmH2O(20), ValveState::CLOSED},
+  });
+}
+
+TEST_F(BlowerFsmTest, TurnOff) {
+  VentParams p_on = VentParams_init_zero;
+  p_on.mode = VentMode_PRESSURE_CONTROL;
+  // 20 breaths/min = 3s/breath.  I:E = 2 means 2s for inspire, 1s for expire.
+  p_on.breaths_per_min = 20;
+  p_on.inspiratory_expiratory_ratio = 2;
+  p_on.peep_cm_h2o = 10;
+  p_on.pip_cm_h2o = 20;
+
+  VentParams p_off = VentParams_init_zero;
+
+  testSequence({
+      {p_off, /*blower_enabled=*/false, 0, cmH2O(0), ValveState::OPEN},
+      {p_on, /*blower_enabled=*/true, 1000, cmH2O(20), ValveState::CLOSED},
+      {p_off, /*blower_enabled=*/false, 1001, cmH2O(0), ValveState::OPEN},
```

Comment:
What's the expected behavior is params change to "vent off" in the middle of a breath, should we still finish the breath? My instinct would be that we shouldn't, because if parameters are for some reason configured real bad (e.g. breaths_per_min = 1), finishing the breath may take a while.

Would be good to add a test case for that, documented either way.

Reply:
> What's the expected behavior is params change to "vent off" in the middle of a breath, should we still finish the breath? My instinct would be that we shouldn't

Agree, that's https://github.com/RespiraWorks/VentilatorSoftware/blob/d8020002bc43fc794bf59613c79cb506104455a0/controller/lib/core/blower_fsm.cpp#L185 and #220.  (I can write the tests now if you want; mostly trying to leave a few good first bugs open.)